### PR TITLE
[ISV-4503] Read-config task supports FBC option

### PR DESCRIPTION
**Pull request by mantomas**
Task is added to hosted and release ISV pipelines. Pipeline will fail, if FBC is enabled and update graph is defined at the same time.

JIRA: ISV-4503

### CODE REVIEW

File: ansible/roles/operator-pipeline/templates/openshift/tasks/read-config.yaml [hunk 1]
```diff
@@ -32,16 +34,30 @@ spec:
...
-
-        # Remove the "-mode" suffix
-        UPGRADE_GRAPH_MODE=`echo $UPGRADE_GRAPH_MODE | sed 's/-mode$//'`
+        # Parse the fbc enablement status
+        FBC_ENABLEMENT=`cat $CONFIG_PATH | yq -r '.fbc.Enabled // "false"'`
+
+        if [ "$FBC_ENABLEMENT" == "true" ]; then
+            if [ "$(cat $CONFIG_PATH | yq -r '.updateGraph')" != "null" ]; then
+                echo "FBC is enabled, updateGraph cannot be defined."
+                exit 1
```
Comment (mporrato) [inline ansible/roles/operator-pipeline/templates/openshift/tasks/read-config.yaml:50]: The AC doesn't say that upgradeGraph is forbidden: it says it's not required.

I have no strong preference for either one: just pointing out the mismatch.

File: ansible/roles/operator-pipeline/templates/openshift/pipelines/operator-release-pipeline.yml [hunk 2]
```diff
@@ -214,6 +214,26 @@ spec:
           workspace: repository
           subPath: base
 
+    - name: read-config
+      taskRef:
+        name: read-config
+        kind: Task
+      runAfter:
+        - detect-changes
+      when: &bundleAdded
```
[the remark this answers is not in the document]
Comment (mantomas) [reply ansible/roles/operator-pipeline/templates/openshift/pipelines/operator-release-pipeline.yml:223]: This logic is [copied](https://github.com/redhat-openshift-ecosystem/operator-pipelines/blob/f3b37b11b97a13afdaf110dbac483b3d8e81d66e/ansible/roles/operator-pipeline/templates/openshift/pipelines/community-release-pipeline.yml#L205) from community release pipeline. I can remove this condition for ISV release pipeline and keep it the same way as hosted.

File: ansible/roles/operator-pipeline/templates/openshift/tasks/read-config.yaml [hunk 3]
```diff
@@ -32,16 +34,30 @@ spec:
...
 
         cat $CONFIG_PATH
 
-        # Parse the upgrade graph mode with default value "replaces"
-        UPGRADE_GRAPH_MODE=`cat $CONFIG_PATH | yq -r '.updateGraph // "replaces"'`
-
-        # Remove the "-mode" suffix
-        UPGRADE_GRAPH_MODE=`echo $UPGRADE_GRAPH_MODE | sed 's/-mode$//'`
+        # Parse the fbc enablement status
+        FBC_ENABLEMENT=`cat $CONFIG_PATH | yq -r '.fbc.Enabled // "false"'`
```
[the remark this answers is not in the document]
Comment (mantomas) [reply ansible/roles/operator-pipeline/templates/openshift/tasks/read-config.yaml]: Fixed, thanks for pointing out.

File: ansible/roles/operator-pipeline/templates/openshift/tasks/read-config.yaml (hunk 1 above)
Comment (mantomas) [reply ansible/roles/operator-pipeline/templates/openshift/tasks/read-config.yaml:50]: This decision is based on offline discussion with @Allda I think this is more clear, not to let users create accidental mismatch providing both options. But also have no strong opinion and can simplify this just to read the config values if anyone have some reasoning to dismiss current implementation.

File: ansible/roles/operator-pipeline/templates/openshift/pipelines/operator-release-pipeline.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Allda) [reply ansible/roles/operator-pipeline/templates/openshift/pipelines/operator-release-pipeline.yml:223]: I see what is happening here... I think we need another jira to make this task compatible with FBC changes and support a use case when no bundle is added. I'll create one and we can groom it later

**Review by mporrato — APPROVED** [review]

**Pull request merged**